lib/customisation.nix

```diff
@@ -312,4 +319,68 @@ rec {
       };
     in self;
 
+  /* Define a new builder supporting recursive attributes (fixed-point style input)
+     from an existing one with such support.
```

Comment:
"Recursive attributes" does sound like "recursive set", and the latter means the `rec { }` syntax we would like to avoid.

Would there be a (preferably short) name to refer to the `builder (finalAttrs { })` pattern?